accelerometer/summariseEpoch.py

```diff
@@ -405,8 +413,91 @@ def calculateECDF(e, inputCol, summary):
         summary[inputCol + '-ecdf-' + str(accUtils.formatNum(x,0)) + 'mg'] = \
             accUtils.formatNum(ecdf, 5)
 
+def calculatePSD(e, summary):
+    """Calculate the power spectral density from fourier analysis of a 1 day frequency
+    
+    :param pandas.DataFrame e: Pandas dataframe of epoch data
+    :param dict summary: Output dictionary containing all summary metrics
 
+    :return: Write dict <summary> keys 'PSD-<W/Hz>'
+    """
+    # collects the sleep column from data frame assumes sleep if it is the highest imputed value
+    y = (np.argmax(e[['sleepImputed', 'moderateImputed', 'sedentaryImputed',
+                       'tasks-lightImputed', 'walkingImputed']].values, axis=1) == 0).astype('int')*2-1
+    n = len(y)
+    k = len(y)*30/(60*60*24)
+    e = -2.j * np.pi * k * np.arange(n) / n
+    # finds the power spectral density for a one day cycle using frouier analysis 
+    res = np.sum(np.exp(e) * y, axis=-1)/n
+    PSD = np.abs(res)**2
+    summary['PSD'] = PSD
+
+def calculateFourierFreq(e, summary):
+    """Calculate the most prevalent frequency in a fourier analysis 
+    
+    :param pandas.DataFrame e: Pandas dataframe of epoch data
+    :param dict summary: Output dictionary containing all summary metrics
 
+    :return: Write dict <summary> keys 'fourier frequency-<1/days>'
+    """
+    y = (np.argmax(e[['sleepImputed', 'moderateImputed', 'sedentaryImputed',
+                       'tasks-lightImputed', 'walkingImputed']].values, axis=1) == 0).astype('int')*2-1
+    # fast fourier transform of the sleep column 
+    fft_y = np.abs(fftpack.fft(y))
+    
+    i =  np.arange(1,len(fft_y)) 
+    k_max = np.argmax(fft_y[i]) + 1
+    n = len(y)
+    # maximises the fourier transform function using the fft_y as a first esitmate 
+    res = sp.optimize.minimize_scalar(lambda k: -np.abs(np.sum(np.exp(-2.j * np.pi * k * np.arange(n) / n) * y, axis=-1)/n),
+                                      bracket = (k_max-1,k_max+1)) 
+    #adjusts the frequency to have the units 1/days
+    freq_mx = float(res.x)/(len(y)*30/(60*60*24))
+    summary['fourier-frequency'] = freq_mx
+    
+def calculateM10L5(e, summary):
+    """Calculates the M10 L5 relatice amplitude from the average acceleration from
+    the ten most active hours and 5 least most active hours 
+    
+    :param pandas.DataFrame e: Pandas dataframe of epoch data
+    :param dict summary: Output dictionary containing all summary metrics
+
+    :return: Write dict <summary> keys 'M10 L5-<rel amp>'
+    """
+    TEN_HOURS = 10*60*2
```

Comment:
Again remove hard coding i.e. instead of using '2', use 'epochPeriod' appropriately instead